src/Library.php

```diff
@@ -250,9 +249,14 @@ public static function process_legacy_payment( PaymentContext $context, PaymentR
 		// Restore $_POST data.
 		$_POST = $post_data;
 
+		// Clear notices so they don't show up in the block.
+		wc_clear_notices();
```

Comment:
Note. I have a PR somewhere I need to raise to capture and surface these notices instead of discarding them.

Reply:
So does this need an issue created so it's not missed or is that something I can just trust you'll follow-up on?

I had considered just capturing the notice instead of hooking in to grab _specific_ payment method errors (as done in this pull) but then I realized that the Woo notice api has no way to differentiate between payment related notices and other type of notices (only whether it's error, success etc). So even _if_ we capture notices, I'm concerned the utility for our use-case might be limited because we can't pipe notices to the proper contexts (payment, shipping, coupons, general checkout etc).

Reply:
I'll create a new PR - I've lost the code 🤦